AcademySoftwareFoundation/MaterialX
# Propagate uv scale and offset

Fixes #1716

Propagate internally added ports when found inside a NodeGraph implementation of a NodeDef.

Testing:
- Run render GLSL tests with the shader write option enabled.
- Open `build/bin/resources/Materials/TestSuite/stdlib/color_management/filename_cm_test/filename_CM_Test_ps.glsl`
- See that all three image nodes (bl, bl1, and tr1) have uv_scale and uv_offset uniform parameters
- See that both `NG_tm_test()` and `NG_tm_retest()` now have scale and offset parameters and that these parameters are passed to the `mx_image_color3()` function.
- See that in `main()` we use all scale and offset uniform parameters when calling either the NodeGraph functions or the regular image node.
[Filename_CM_Test_ps.zip](https://github.com/AcademySoftwareFoundation/MaterialX/files/15461905/Filename_CM_Test_ps.zip)
